### PR TITLE
test: test the dashboard admins property

### PR DESCRIPTION
**Pull request by thomasheartman**
This PR adds tests for the new admins property of the personal dashboard API payload.

It checks that only user admins are added and that their image URL is not an empty string. In doing this, also fixes an issue where the image URL wouldn't be generated correctly.

## Discussion points

Some of the test feels like it might be better testing on a deeper level (i.e. the account store). However, from an initial glance, I think that would require more setup and work, so I'm leaving it in the dashboard test for now as that's where it's ultimately useful. But we can discuss if we should move it.

### CODE REVIEW

File: src/lib/types/user.ts [hunk 1]
```diff
@@ -33,7 +33,7 @@ export interface IUser {
     scimId?: string;
 }
 
-export type MinimalUser = Pick<
+export type MinimalUiUser = Pick<
```
Comment (kwasniew) [inline src/lib/types/user.ts:36]: What if this data is accessed through the API? Would it still make sense to call it "Ui"

File: src/lib/features/personal-dashboard/personal-dashboard-controller.e2e.test.ts [hunk 2]
```diff
@@ -217,3 +217,52 @@ test('should return personal dashboard project details', async () => {
...
+        .returning('*');
+    await app.services.accessService.setUserRootRole(
+        serviceAdminId,
+        adminRoleId,
+    );
+
+    const { body } = await app.request.get(`/api/admin/personal-dashboard`);
+
+    const admins = body.admins;
+    admins.sort((a, b) => a.id - b.id);
```
Comment (kwasniew) [inline src/lib/features/personal-dashboard/personal-dashboard-controller.e2e.test.ts:254]: I'm not a big fan of sorting in e2e/component tests. Is it changing between test runs?

**Review by kwasniew — APPROVED** [review]

File: src/lib/types/user.ts (hunk 1 above)
Comment (thomasheartman) [reply src/lib/types/user.ts:36]: good point. The data is intended to be used to render a user (so it includes the image url, notably). I don't mind leaving or removing that part of the name. Or if you have a different suggestion, I'm very open to that too.

File: src/lib/features/personal-dashboard/personal-dashboard-controller.e2e.test.ts (hunk 2 above)
Comment (thomasheartman) [reply src/lib/features/personal-dashboard/personal-dashboard-controller.e2e.test.ts:254]: no idea. why don't you like sorting the data in tests like this? i'm thinking it'll keep it from breaking regardless of any changes that we do on the back end. We can leave it out, of course, but it'd be annoying if it suddenly broke later when we haven't specified the orde.r

File: src/lib/features/personal-dashboard/personal-dashboard-controller.e2e.test.ts [hunk 3]
```diff
@@ -217,3 +217,52 @@ test('should return personal dashboard project details', async () => {
...
+    const admin2 = await userService.createUser({
+        username: 'John',
+        name: 'John Admin',
+        rootRole: adminRoleId,
+    });
+
+    // service account that shouldn't be listed in the output. Service
+    // accounts are enterprise functionality, so there's no service to
+    // call here
+    const [{ id: serviceAdminId }] = await db
```
[the remark this answers is not in the document]
Comment (thomasheartman) [reply src/lib/features/personal-dashboard/personal-dashboard-controller.e2e.test.ts:239]: The controller, service, and store that create service accounts are all in the enterprise repo, if that's what you're asking?

File: src/lib/types/user.ts (hunk 1 above)
Comment (kwasniew) [reply src/lib/types/user.ts:36]: Personally I'd stick to the previous name

Comment (thomasheartman) [reply src/lib/types/user.ts]: Done ✅

File: src/lib/features/personal-dashboard/personal-dashboard-controller.e2e.test.ts (hunk 2 above)
Comment (thomasheartman) [reply src/lib/features/personal-dashboard/personal-dashboard-controller.e2e.test.ts]: be151d1851dc0445d30a1b1193ac93fa647df85c

**Pull request merged**